src/reporter/comments-reporter.ts

```diff
@@ -0,0 +1,170 @@
+/*
+   Copyright 2022 Mitch Spano
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+	 https://www.apache.org/licenses/LICENSE-2.0
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+ */
+
+import { getGithubFilePath, getScannerViolationType } from "../common";
+
+import core from "@actions/core";
+import { Octokit } from "@octokit/action";
+import { context } from "@actions/github";
+import {
+  BaseReporter,
+  GithubComment,
+  GithubExistingComment,
+} from "./reporter.types";
+import { ScannerViolation } from "../sfdxCli";
+
+const HIDDEN_COMMENT_PREFIX = "<!--sfdx-scanner-->";
+
+export class CommentsReporter extends BaseReporter<GithubComment> {
+  private performGithubRequest<T>(
```

Comment:
It seems like both the `annotations` and the `comments` reporters need to generate an instance of octokit and perform a  GitHub request. Is it possible for us to move any of logic or method signature declaration up to the parent abstract class?

Reply:
Hmmm, I think trying to abstract it away is not going to simplify much. Octokit is designed in this way for a reason, a lot of the endpoints are pretty unique. If you feel strongly I can put something together, but I'm not sure what we could do that would be an actual improvement